node/forest_libp2p/src/service.rs

```diff
@@ -410,271 +410,292 @@ async fn handle_network_message<P: StoreParams>(
+        }
+        BitswapEvent::Complete(query_id, result) => match result {
+            Ok(()) => {
+                let prefix = get_prefix(&query_id);
+                debug!("{prefix} bitswap query {query_id} completed successfully");
+                if let Some(cid) = outgoing_bitswap_query_ids.remove(&query_id) {
+                    emit_event(
+                        network_sender_out,
+                        NetworkEvent::BitswapResponseInbound { query_id, cid },
+                    )
+                    .await;
+                }
             }
             Err(err) => {
-                let err = err.to_string();
-                let peer = ping_event.peer.to_base58();
-                warn!("{err}: {peer}",);
-                if err.contains("protocol not supported") {
-                    warn!("Banning peer {peer} due to protocol error");
-                    swarm.ban_peer_id(ping_event.peer);
+                let prefix = get_prefix(&query_id);
+                let msg = format!("{prefix} bitswap query {query_id} completed with error: {err}");
+                if outgoing_bitswap_query_ids.contains_key(&query_id) {
+                    warn!("{msg}");
+                } else {
+                    debug!("{msg}");
                 }
             }
         },
-        ForestBehaviourEvent::Identify(_) => {}
-        ForestBehaviourEvent::ChainExchange(ce_event) => match ce_event {
-            RequestResponseEvent::Message { peer, message } => match message {
+    }
+}
+
+fn handle_ping_event<P: StoreParams>(
+    ping_event: ping::Event,
+    swarm: &mut Swarm<ForestBehaviour<P>>,
+) {
+    match ping_event.result {
+        Ok(ping::Success::Ping { rtt }) => {
+            trace!(
+                "PingSuccess::Ping rtt to {} is {} ms",
+                ping_event.peer.to_base58(),
+                rtt.as_millis()
+            );
+        }
+        Ok(ping::Success::Pong) => {
+            trace!("PingSuccess::Pong from {}", ping_event.peer.to_base58());
+        }
+        Err(ping::Failure::Other { error }) => {
+            warn!(
+                "PingFailure::Other {}: {}",
+                ping_event.peer.to_base58(),
+                error
+            );
+        }
+        Err(err) => {
+            let err = err.to_string();
+            let peer = ping_event.peer.to_base58();
+            warn!("{err}: {peer}",);
+            if err.contains("protocol not supported") {
+                warn!("Banning peer {peer} due to protocol error");
+                swarm.ban_peer_id(ping_event.peer);
+            }
+        }
+    }
+}
+
+async fn handle_chain_exchange_event<DB, P: StoreParams>(
+    ce_event: RequestResponseEvent<ChainExchangeRequest, ChainExchangeResponse>,
+    db: &Arc<ChainStore<DB>>,
+    network_sender_out: &flume::Sender<NetworkEvent>,
+    cx_request_table: &mut HashMap<
+        RequestId,
+        futures::channel::oneshot::Sender<Result<ChainExchangeResponse, RequestResponseError>>,
```

Comment:
```suggestion
        OneShotSender<Result<ChainExchangeResponse, RequestResponseError>>,
```

Reply:
There are several of these. You can update all of them.

Reply:
Done